profile/profile.go

```diff
@@ -715,6 +715,34 @@ func (s *Sample) HasLabel(key, value string) bool {
 	return false
 }
 
+// SetNumLabel sets the specified key to the specified value for all samples in the
+// profile. "unit" is a slice that describes the units that each corresponding member
+// of "values" is measured in (e.g. bytes or seconds).  If there is no relevant
+// unit for a given value, that member of "unit" should be the empty string.
```

Comment:
I would add a note that `unit` must either be nil or equal in length to `value`.

Added a few clarifying comments in this in https://github.com/google/pprof/pull/750.

Reply:
Done.